packages/sdk/src/dataProtector/sharing/removeProtectedDataForSale.ts

```diff
@@ -0,0 +1,115 @@
+import type { GraphQLClient } from 'graphql-request';
+import { DEFAULT_SHARING_CONTRACT_ADDRESS } from '../../config/config.js';
+import { ErrorWithData } from '../../utils/errors.js';
+import { addressSchema, throwIfMissing } from '../../utils/validators.js';
+import {
+  Address,
+  IExecConsumer,
+  RemoveProtectedDataForSaleParams,
+  SubgraphConsumer,
+  SuccessWithTransactionHash,
+} from '../types/index.js';
+import { getSharingContract } from './smartContract/getSharingContract.js';
+import { getProtectedDataById } from './subgraph/getProtectedDataById.js';
+
+export const removeProtectedDataForSale = async ({
+  iexec = throwIfMissing(),
+  graphQLClient = throwIfMissing(),
+  protectedDataAddress,
+}: IExecConsumer &
+  SubgraphConsumer &
+  RemoveProtectedDataForSaleParams): Promise<SuccessWithTransactionHash> => {
+  addressSchema()
+    .required()
+    .label('protectedDataAddress')
+    .validateSync(protectedDataAddress);
```

Comment:
Little change here compared to existing code: I'm not created a new variable like `vProtectedDataAddress`. It looks like it's not necessary and validation will work as expected.